architecture/2025-10-terraform-bicep-settings.md

```diff
@@ -0,0 +1,325 @@
+
+## Terms and Definitions
+
+- **TerraformSettings**: New `Radius.Core/terraformSettings` resource encapsulating `.terraformrc`, backend, environment, and logging settings. Migrates everything currently in `recipeConfig.terraform` (provider mirrors/credentials, backend blocks, `env` variables, trace logging flags) while shifting env secret injection to recipe parameters.
+- **BicepSettings**: New `Radius.Core/bicepSettings` resource describing registry authentication. Carries forward the existing `recipeConfig.bicep.authentication` map (registry host → secret ID); no additional Bicep settings exist today.
+- **Installer Async Handler**: Implementation registered with the existing async worker service (`pkg/server/asyncworker.go`) that consumes install/uninstall queue messages (`pkg/components/queue`) and manages Terraform binaries.
+
+## Objectives
+
+> **Issue Reference:** <https://github.com/radius-project/radius/issues/10615>
+
+### Goals
+
+**Terraform**
+
+- Add `Radius.Core/terraformSettings` resources and wire `Radius.Core/environments` (the new environment type) to reference reusable configuration.
+- Introduce an installer async pipeline and CLI-driven Terraform binary lifecycle (`rad terraform install|uninstall|status`) with operator control over version, source URL, and checksum.
+- Allow Terraform settings (provider mirrors, credentials, env vars, backend blocks) to flow through unchanged so Radius is unopinionated about Terraform configuration.
+- Deliver Phase 1 with the existing Kubernetes backend, and stage AzureRM/S3 backend support as Phase 2 follow-up while keeping Tier-2 backends on the roadmap (for example `oss`, `gcs`, `http`, `oci`, `pg`, `cos`).
+
+**Bicep**
+
+- Add `bicepSettings` resources so registry authentication is reusable outside `recipeConfig`.
+- Preserve support for BasicAuth, Azure workload identity, and AWS IRSA secret injection with no new runtime behaviors.
+
+### Non Goals
+
+- Delivering `terraform plan` functionality (covered by a future spec).
+- Enabling non-Kubernetes Terraform backends in Phase 1. AzureRM and S3 land in Phase 2; the rest of the Tier-2 backlog remains out of scope here.
+- Reworking Bicep execution beyond registry authentication parity.
+- Adding new capabilities for legacy `Applications.Core/environments`.
+- Changing recipe parameter or SecretStore semantics; env secret injection continues via recipe parameters.
+- Modifying Bicep runtime behavior or the bundled Bicep CLI; only registry authentication moves into `bicepSettings`.
```

Comment:
Should we implement bicep installation in the same way that Terraform installation is specified in this design? Or plan the work for future?